auth/tokenizer.go

```diff
@@ -0,0 +1,135 @@
+package auth
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"os"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/pborman/uuid"
+	"gitlab.com/LUSHDigital/soa/first-class/api-gateway/service/config"
+)
+
+// Tokeniser is the auth tokeniser for JSON Web Tokens
+type Tokeniser struct {
+	privateKey *rsa.PrivateKey
+	publicKey  *rsa.PublicKey
+	authIssuer string
+}
+
+// NewMockTokeniser creates a new tokeniser with a random key pair
+func NewMockTokeniser() (*Tokeniser, error) {
+	reader := rand.Reader
+	bitSize := 2048
+	privateKey, err := rsa.GenerateKey(reader, bitSize)
+	if err != nil {
+		return nil, err
+	}
+	publicKey := &privateKey.PublicKey
+	name, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+	return &Tokeniser{
+		privateKey: privateKey,
+		publicKey:  publicKey,
+		authIssuer: name,
+	}, nil
+}
+
+// NewPublicTokeniser returns a new JWT instance
+func NewPublicTokeniser(bPublicKey string) (*Tokeniser, error) {
+	publicKey, err := jwt.ParseRSAPublicKeyFromPEM([]byte(bPublicKey))
+	if err != nil {
+		return nil, err
+	}
+	return &Tokeniser{publicKey: publicKey}, nil
+}
+
+// NewTokeniser returns a new JWT instance
+func NewTokeniser(bPrivateKey, bPublicKey, issuer string) (*Tokeniser, error) {
```

Comment:
I'd prefer to see `bPrivateKey` and `bPublicKey` to be byte slices, to play nicely with `crypto/x509`.

Or even one better, `*rsa.PublicKey` and `*rsa.PrivateKey`

Reply:
I think this is just a matter of when the conversion happens. They will come in as strings from the env vars and get turned into []bytes and the *rsa.things. It just depends when in the method chain that conversion happens.